cli/build.go

```diff
@@ -147,15 +119,54 @@ func (cmd *BuildFlags) postInit() error {
 		}
 	}
 
+	// Init the image store.
+	imageStore, err := storage.NewImageStore(cmd.StorageDir)
+	if err != nil {
+		return fmt.Errorf("failed to init image store: %s", err)
+	}
+	cmd.imageStore = imageStore
+
 	// Verify storage dir is not child of internal dir.
 	if pathutils.IsDescendantOfAny(cmd.StorageDir, []string{pathutils.DefaultInternalDir}) {
 		return fmt.Errorf("storage dir cannot be under internal dir %s",
 			pathutils.DefaultInternalDir)
 	}
+	return nil
+}
 
+func (cmd BuildFlags) initRegistryGlobals() error {
+	if cmd.RegistryConfig == "" {
+		// TODO(pourchet): Shouldn't we do this regardless of if a registry config was passed?
```

Comment:
well map doesnt have an order and .* is dangerous...

Reply:
That's fair